src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java

```diff
@@ -117,6 +123,19 @@ public boolean hasDuplicate(BibEntry entry) {
                 .containsDuplicate(selectedDb.getValue().getDatabase(), entry, selectedDb.getValue().getMode()).isPresent();
     }
 
+    public String getSourceString(BibEntry entry) {
+        StringWriter writer = new StringWriter();
+        BibWriter bibWriter = new BibWriter(writer, OS.NEWLINE);
+        FieldWriter fieldWriter = FieldWriter.buildIgnoreHashes(preferences.getFieldPreferences());
+        try {
+            new BibEntryWriter(fieldWriter, entryTypesManager).write(entry, bibWriter, selectedDb.getValue().getMode());
+        } catch (
+                IOException ioException) {
+            return null;
```

Comment:
In Java, one should avoid `null`, because this leads to NullPointerExceptions, which are hard to debug.

In your case, you can fallback to a default value - the empty string.

```suggestion
            return "";
```

You could also include the erorr message, but I think, this is not helpful for the user.